adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/JDBCSpace.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2023 nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.nosqlbench.adapter.jdbc;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import io.nosqlbench.api.config.standard.ConfigModel;
+import io.nosqlbench.api.config.standard.NBConfigModel;
+import io.nosqlbench.api.config.standard.NBConfiguration;
+import io.nosqlbench.api.config.standard.Param;
+import io.nosqlbench.api.errors.OpConfigError;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.sql.Connection;
+import java.util.Optional;
+
+public class JDBCSpace implements AutoCloseable {
+    private final static Logger logger = LogManager.getLogger(JDBCSpace.class);
+    private final String spaceName;
+    private HikariConfig hikariConfig;
+    private HikariDataSource hikariDataSource;
+    private Connection connection;
+
+    public JDBCSpace(String spaceName, NBConfiguration cfg) {
+        this.spaceName = spaceName;
+        this.hikariDataSource = createClient(cfg);
+    }
+
+    public Connection getConnection() {
+        return this.connection;
+    }
+
+    public HikariDataSource getHikariDataSource() {
+        return this.hikariDataSource;
+    }
+
+    private HikariDataSource createClient(NBConfiguration cfg) {
+        hikariConfig = new HikariConfig();
+
+        Optional<String> url = cfg.getOptional("url");
+        if (url.isEmpty()) {
+            throw new OpConfigError("url option is required.");
```

Comment:
For required params, you should simply set them in the config model and let it do the lifting. If you specify default, it is implied as required, but you can also flag required options without setting defaults for them.

Reply:
This is now addressed. 👍🏼